src/hyperion/experiment_plans/flyscan_xray_centre_plan.py

```diff
@@ -232,27 +233,32 @@ def run_gridscan_and_move(
 
     yield from setup_zebra_for_gridscan(fgs_composite.zebra)
 
-    hyperion.log.LOGGER.info("Starting grid scan")
+    LOGGER.info("Starting grid scan")
 
     yield from run_gridscan(fgs_composite, parameters)
 
-    # the data were submitted to zocalo by the zocalo callback during the gridscan,
-    # but results may not be ready, and need to be collected regardless.
-    # it might not be ideal to block for this, see #327
-    xray_centre, bbox_size = subscriptions.zocalo_handler.wait_for_results(initial_xyz)
+    yield from trigger_zocalo(fgs_composite.zocalo)
+    res = yield from get_processing_results(fgs_composite.zocalo)
+    xray_centre, bbox_size = res
 
-    if bbox_size is not None:
+    if xray_centre is not None and bbox_size is not None:
+        xray_centre = parameters.experiment_params.grid_position_to_motor_position(
+            xray_centre
+        )
         with TRACER.start_span("change_aperture"):
             yield from set_aperture_for_bbox_size(
                 fgs_composite.aperture_scatterguard, bbox_size
             )
+    else:
+        LOGGER.warning
+        xray_centre = initial_xyz
```

Comment:
Must: `LOGGER.warning` doesn't do much... Also, I think we probably want to do this on exception too